cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -535,6 +541,166 @@ public void programSpecification(HttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, specification);
   }
 
+  @PUT
```

Comment:
Even though this returns a 409 if a schedule already exists, I believe that this should be a POST because adding a new schedules isn't an idempotent operation.

Up for discussion: @poornachandra @anew